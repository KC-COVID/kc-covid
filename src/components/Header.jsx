import React from 'react';
import { Navbar, Nav } from 'react-bootstrap';
import { useHistory, useLocation } from 'react-router-dom';
import logo from '../assets/KcCovidLogo_color.png';

/**
 * The header component for the page. Handles rendering the tab bar and routing to the proper page on tab click.
 */
function Header() {
  // react-router hooks. History let's us navigate to new and previous routes. Location is our current route.
  const history = useHistory();
  const location = useLocation();

  const path = location.pathname;

  return (
    <Navbar bg="light" fixed="top" expand="lg">
      <Navbar.Brand onClick={() => history.push('/')}>
        <img src={logo} style={{ height: '20px', marginTop: '-8px' }} alt="logo" />
      </Navbar.Brand>
      <Navbar.Toggle aria-controls="responsive-navbar-nav" />
      <Navbar.Collapse id="responsive-navbar-nav">
        <Nav className="mr-auto">
          <Nav.Link onClick={() => history.push('/')} id="home-tab" className={path === '/' ? 'active' : ''}>Home</Nav.Link>
          <Nav.Link
            onClick={() => history.push('/volunteer')}
            id="volunteer-tab"
            className={path === '/volunteer' ? 'active' : ''}
          >
            Volunteer
          </Nav.Link>
          <Nav.Link
            onClick={() => history.push('/aid')}
            id="aid-tab"
            className={path === '/aid' ? 'active' : ''}
          >
            Request Aid
          </Nav.Link>
          <Nav.Link
            onClick={() => history.push('/submit')}
            id="submit-tab"
            className={path === '/submit' ? 'active' : ''}
          >
            Submit Data
          </Nav.Link>
          <Nav.Link
            onClick={() => history.push("/resources")}
            id="submit-tab"
            className={path === '/submit' ? 'active' : ''}
          >
            Resources
          </Nav.Link>
        </Nav>
      </Navbar.Collapse>
    </Navbar>
  );
}

export default Header;
