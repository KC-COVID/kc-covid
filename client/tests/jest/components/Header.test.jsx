import React from 'react';
import { shallow } from 'enzyme';

import Header from '../../../src/components/Header';

const mockPush = jest.fn();
let mockPathName = '/';

jest.mock('react-router-dom', () => ({
  useHistory: () => ({ push: mockPush }),
  useLocation: () => ({ pathname: mockPathName }),
}));

describe('Header', () => {
  afterEach(() => {
    mockPush.mockReset();
    mockPathName = '/';
  });

  it('renders the nav bar with a brand, toggle, and collapse', () => {
    const shallowedHeader = shallow(<Header />);
    expect(shallowedHeader).toMatchSnapshot();
  });

  it('navigates to the home route when the home tab is clicked', () => {
    const shallowedHeader = shallow(<Header />);

    const homeTab = shallowedHeader.find('#home-tab');
    homeTab.simulate('click', {});

    expect(mockPush).toHaveBeenCalledWith('/');
    expect(shallowedHeader).toMatchSnapshot();
  });

  it('navigates to the volunteer route when the volunteer tab is clicked', () => {
    const shallowedHeader = shallow(<Header />);

    const volunteerTab = shallowedHeader.find('#volunteer-tab');
    volunteerTab.simulate('click', {});

    expect(mockPush).toHaveBeenCalledWith('/volunteer');
    expect(shallowedHeader).toMatchSnapshot();
  });

  it('navigates to the aid route when the aid tab is clicked', () => {
    const shallowedHeader = shallow(<Header />);

    const aidTab = shallowedHeader.find('#aid-tab');
    aidTab.simulate('click', {});

    expect(mockPush).toHaveBeenCalledWith('/aid');
    expect(shallowedHeader).toMatchSnapshot();
  });

  it('navigates to the submit route when the submit tab is clicked', () => {
    const shallowedHeader = shallow(<Header />);

    const submitTab = shallowedHeader.find('#submit-tab');
    submitTab.simulate('click', {});

    expect(mockPush).toHaveBeenCalledWith('/submit');
    expect(shallowedHeader).toMatchSnapshot();
  });

  it('displays the home tab as active when the path is /', () => {
    const shallowedHeader = shallow(<Header />);

    const homeTab = shallowedHeader.find('#home-tab');

    expect(homeTab.hasClass('active')).toBeTruthy();
    expect(shallowedHeader).toMatchSnapshot();
  });

  it('displays the volunteer tab as active when the path is /volunteer', () => {
    mockPathName = '/volunteer';
    const shallowedHeader = shallow(<Header />);

    const homeTab = shallowedHeader.find('#home-tab');
    const volunteerTab = shallowedHeader.find('#volunteer-tab');

    expect(homeTab.hasClass('active')).toBeFalsy();
    expect(volunteerTab.hasClass('active')).toBeTruthy();
    expect(shallowedHeader).toMatchSnapshot();
  });

  it('displays the aid tab as active when the path is /aid', () => {
    mockPathName = '/aid';
    const shallowedHeader = shallow(<Header />);

    const homeTab = shallowedHeader.find('#home-tab');
    const aidTab = shallowedHeader.find('#aid-tab');

    expect(homeTab.hasClass('active')).toBeFalsy();
    expect(aidTab.hasClass('active')).toBeTruthy();
    expect(shallowedHeader).toMatchSnapshot();
  });

  it('displays the submit tab as active when the path is /submit', () => {
    mockPathName = '/submit';
    const shallowedHeader = shallow(<Header />);

    const homeTab = shallowedHeader.find('#home-tab');
    const aidTab = shallowedHeader.find('#submit-tab');

    expect(homeTab.hasClass('active')).toBeFalsy();
    expect(aidTab.hasClass('active')).toBeTruthy();
    expect(shallowedHeader).toMatchSnapshot();
  });
});
