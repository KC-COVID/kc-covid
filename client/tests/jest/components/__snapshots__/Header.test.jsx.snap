// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header displays the aid tab as active when the path is /aid 1`] = `
<Navbar
  bg="light"
  collapseOnSelect={false}
  expand="lg"
  fixed="top"
  variant="light"
>
  <NavbarBrand
    onClick={[Function]}
  >
    <img
      alt="logo"
      src={null}
      style={
        Object {
          "height": "20px",
          "marginTop": "-8px",
        }
      }
    />
  </NavbarBrand>
  <NavbarToggle
    aria-controls="responsive-navbar-nav"
    label="Toggle navigation"
  />
  <NavbarCollapse
    id="responsive-navbar-nav"
  >
    <Nav
      className="mr-auto"
      fill={false}
      justify={false}
    >
      <NavLink
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "SafeAnchor",
            "render": [Function],
          }
        }
        className=""
        disabled={false}
        id="home-tab"
        onClick={[Function]}
      >
        Home
      </NavLink>
      <NavLink
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "SafeAnchor",
            "render": [Function],
          }
        }
        className=""
        disabled={false}
        id="volunteer-tab"
        onClick={[Function]}
      >
        Volunteer
      </NavLink>
      <NavLink
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "SafeAnchor",
            "render": [Function],
          }
        }
        className="active"
        disabled={false}
        id="aid-tab"
        onClick={[Function]}
      >
        Request Aid
      </NavLink>
      <NavLink
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "SafeAnchor",
            "render": [Function],
          }
        }
        className=""
        disabled={false}
        id="submit-tab"
        onClick={[Function]}
      >
        Submit Data
      </NavLink>
    </Nav>
  </NavbarCollapse>
</Navbar>
`;

exports[`Header displays the home tab as active when the path is / 1`] = `
<Navbar
  bg="light"
  collapseOnSelect={false}
  expand="lg"
  fixed="top"
  variant="light"
>
  <NavbarBrand
    onClick={[Function]}
  >
    <img
      alt="logo"
      src={null}
      style={
        Object {
          "height": "20px",
          "marginTop": "-8px",
        }
      }
    />
  </NavbarBrand>
  <NavbarToggle
    aria-controls="responsive-navbar-nav"
    label="Toggle navigation"
  />
  <NavbarCollapse
    id="responsive-navbar-nav"
  >
    <Nav
      className="mr-auto"
      fill={false}
      justify={false}
    >
      <NavLink
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "SafeAnchor",
            "render": [Function],
          }
        }
        className="active"
        disabled={false}
        id="home-tab"
        onClick={[Function]}
      >
        Home
      </NavLink>
      <NavLink
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "SafeAnchor",
            "render": [Function],
          }
        }
        className=""
        disabled={false}
        id="volunteer-tab"
        onClick={[Function]}
      >
        Volunteer
      </NavLink>
      <NavLink
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "SafeAnchor",
            "render": [Function],
          }
        }
        className=""
        disabled={false}
        id="aid-tab"
        onClick={[Function]}
      >
        Request Aid
      </NavLink>
      <NavLink
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "SafeAnchor",
            "render": [Function],
          }
        }
        className=""
        disabled={false}
        id="submit-tab"
        onClick={[Function]}
      >
        Submit Data
      </NavLink>
    </Nav>
  </NavbarCollapse>
</Navbar>
`;

exports[`Header displays the submit tab as active when the path is /submit 1`] = `
<Navbar
  bg="light"
  collapseOnSelect={false}
  expand="lg"
  fixed="top"
  variant="light"
>
  <NavbarBrand
    onClick={[Function]}
  >
    <img
      alt="logo"
      src={null}
      style={
        Object {
          "height": "20px",
          "marginTop": "-8px",
        }
      }
    />
  </NavbarBrand>
  <NavbarToggle
    aria-controls="responsive-navbar-nav"
    label="Toggle navigation"
  />
  <NavbarCollapse
    id="responsive-navbar-nav"
  >
    <Nav
      className="mr-auto"
      fill={false}
      justify={false}
    >
      <NavLink
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "SafeAnchor",
            "render": [Function],
          }
        }
        className=""
        disabled={false}
        id="home-tab"
        onClick={[Function]}
      >
        Home
      </NavLink>
      <NavLink
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "SafeAnchor",
            "render": [Function],
          }
        }
        className=""
        disabled={false}
        id="volunteer-tab"
        onClick={[Function]}
      >
        Volunteer
      </NavLink>
      <NavLink
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "SafeAnchor",
            "render": [Function],
          }
        }
        className=""
        disabled={false}
        id="aid-tab"
        onClick={[Function]}
      >
        Request Aid
      </NavLink>
      <NavLink
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "SafeAnchor",
            "render": [Function],
          }
        }
        className="active"
        disabled={false}
        id="submit-tab"
        onClick={[Function]}
      >
        Submit Data
      </NavLink>
    </Nav>
  </NavbarCollapse>
</Navbar>
`;

exports[`Header displays the volunteer tab as active when the path is /volunteer 1`] = `
<Navbar
  bg="light"
  collapseOnSelect={false}
  expand="lg"
  fixed="top"
  variant="light"
>
  <NavbarBrand
    onClick={[Function]}
  >
    <img
      alt="logo"
      src={null}
      style={
        Object {
          "height": "20px",
          "marginTop": "-8px",
        }
      }
    />
  </NavbarBrand>
  <NavbarToggle
    aria-controls="responsive-navbar-nav"
    label="Toggle navigation"
  />
  <NavbarCollapse
    id="responsive-navbar-nav"
  >
    <Nav
      className="mr-auto"
      fill={false}
      justify={false}
    >
      <NavLink
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "SafeAnchor",
            "render": [Function],
          }
        }
        className=""
        disabled={false}
        id="home-tab"
        onClick={[Function]}
      >
        Home
      </NavLink>
      <NavLink
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "SafeAnchor",
            "render": [Function],
          }
        }
        className="active"
        disabled={false}
        id="volunteer-tab"
        onClick={[Function]}
      >
        Volunteer
      </NavLink>
      <NavLink
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "SafeAnchor",
            "render": [Function],
          }
        }
        className=""
        disabled={false}
        id="aid-tab"
        onClick={[Function]}
      >
        Request Aid
      </NavLink>
      <NavLink
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "SafeAnchor",
            "render": [Function],
          }
        }
        className=""
        disabled={false}
        id="submit-tab"
        onClick={[Function]}
      >
        Submit Data
      </NavLink>
    </Nav>
  </NavbarCollapse>
</Navbar>
`;

exports[`Header navigates to the aid route when the aid tab is clicked 1`] = `
<Navbar
  bg="light"
  collapseOnSelect={false}
  expand="lg"
  fixed="top"
  variant="light"
>
  <NavbarBrand
    onClick={[Function]}
  >
    <img
      alt="logo"
      src={null}
      style={
        Object {
          "height": "20px",
          "marginTop": "-8px",
        }
      }
    />
  </NavbarBrand>
  <NavbarToggle
    aria-controls="responsive-navbar-nav"
    label="Toggle navigation"
  />
  <NavbarCollapse
    id="responsive-navbar-nav"
  >
    <Nav
      className="mr-auto"
      fill={false}
      justify={false}
    >
      <NavLink
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "SafeAnchor",
            "render": [Function],
          }
        }
        className="active"
        disabled={false}
        id="home-tab"
        onClick={[Function]}
      >
        Home
      </NavLink>
      <NavLink
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "SafeAnchor",
            "render": [Function],
          }
        }
        className=""
        disabled={false}
        id="volunteer-tab"
        onClick={[Function]}
      >
        Volunteer
      </NavLink>
      <NavLink
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "SafeAnchor",
            "render": [Function],
          }
        }
        className=""
        disabled={false}
        id="aid-tab"
        onClick={[Function]}
      >
        Request Aid
      </NavLink>
      <NavLink
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "SafeAnchor",
            "render": [Function],
          }
        }
        className=""
        disabled={false}
        id="submit-tab"
        onClick={[Function]}
      >
        Submit Data
      </NavLink>
    </Nav>
  </NavbarCollapse>
</Navbar>
`;

exports[`Header navigates to the home route when the home tab is clicked 1`] = `
<Navbar
  bg="light"
  collapseOnSelect={false}
  expand="lg"
  fixed="top"
  variant="light"
>
  <NavbarBrand
    onClick={[Function]}
  >
    <img
      alt="logo"
      src={null}
      style={
        Object {
          "height": "20px",
          "marginTop": "-8px",
        }
      }
    />
  </NavbarBrand>
  <NavbarToggle
    aria-controls="responsive-navbar-nav"
    label="Toggle navigation"
  />
  <NavbarCollapse
    id="responsive-navbar-nav"
  >
    <Nav
      className="mr-auto"
      fill={false}
      justify={false}
    >
      <NavLink
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "SafeAnchor",
            "render": [Function],
          }
        }
        className="active"
        disabled={false}
        id="home-tab"
        onClick={[Function]}
      >
        Home
      </NavLink>
      <NavLink
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "SafeAnchor",
            "render": [Function],
          }
        }
        className=""
        disabled={false}
        id="volunteer-tab"
        onClick={[Function]}
      >
        Volunteer
      </NavLink>
      <NavLink
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "SafeAnchor",
            "render": [Function],
          }
        }
        className=""
        disabled={false}
        id="aid-tab"
        onClick={[Function]}
      >
        Request Aid
      </NavLink>
      <NavLink
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "SafeAnchor",
            "render": [Function],
          }
        }
        className=""
        disabled={false}
        id="submit-tab"
        onClick={[Function]}
      >
        Submit Data
      </NavLink>
    </Nav>
  </NavbarCollapse>
</Navbar>
`;

exports[`Header navigates to the submit route when the submit tab is clicked 1`] = `
<Navbar
  bg="light"
  collapseOnSelect={false}
  expand="lg"
  fixed="top"
  variant="light"
>
  <NavbarBrand
    onClick={[Function]}
  >
    <img
      alt="logo"
      src={null}
      style={
        Object {
          "height": "20px",
          "marginTop": "-8px",
        }
      }
    />
  </NavbarBrand>
  <NavbarToggle
    aria-controls="responsive-navbar-nav"
    label="Toggle navigation"
  />
  <NavbarCollapse
    id="responsive-navbar-nav"
  >
    <Nav
      className="mr-auto"
      fill={false}
      justify={false}
    >
      <NavLink
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "SafeAnchor",
            "render": [Function],
          }
        }
        className="active"
        disabled={false}
        id="home-tab"
        onClick={[Function]}
      >
        Home
      </NavLink>
      <NavLink
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "SafeAnchor",
            "render": [Function],
          }
        }
        className=""
        disabled={false}
        id="volunteer-tab"
        onClick={[Function]}
      >
        Volunteer
      </NavLink>
      <NavLink
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "SafeAnchor",
            "render": [Function],
          }
        }
        className=""
        disabled={false}
        id="aid-tab"
        onClick={[Function]}
      >
        Request Aid
      </NavLink>
      <NavLink
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "SafeAnchor",
            "render": [Function],
          }
        }
        className=""
        disabled={false}
        id="submit-tab"
        onClick={[Function]}
      >
        Submit Data
      </NavLink>
    </Nav>
  </NavbarCollapse>
</Navbar>
`;

exports[`Header navigates to the volunteer route when the volunteer tab is clicked 1`] = `
<Navbar
  bg="light"
  collapseOnSelect={false}
  expand="lg"
  fixed="top"
  variant="light"
>
  <NavbarBrand
    onClick={[Function]}
  >
    <img
      alt="logo"
      src={null}
      style={
        Object {
          "height": "20px",
          "marginTop": "-8px",
        }
      }
    />
  </NavbarBrand>
  <NavbarToggle
    aria-controls="responsive-navbar-nav"
    label="Toggle navigation"
  />
  <NavbarCollapse
    id="responsive-navbar-nav"
  >
    <Nav
      className="mr-auto"
      fill={false}
      justify={false}
    >
      <NavLink
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "SafeAnchor",
            "render": [Function],
          }
        }
        className="active"
        disabled={false}
        id="home-tab"
        onClick={[Function]}
      >
        Home
      </NavLink>
      <NavLink
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "SafeAnchor",
            "render": [Function],
          }
        }
        className=""
        disabled={false}
        id="volunteer-tab"
        onClick={[Function]}
      >
        Volunteer
      </NavLink>
      <NavLink
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "SafeAnchor",
            "render": [Function],
          }
        }
        className=""
        disabled={false}
        id="aid-tab"
        onClick={[Function]}
      >
        Request Aid
      </NavLink>
      <NavLink
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "SafeAnchor",
            "render": [Function],
          }
        }
        className=""
        disabled={false}
        id="submit-tab"
        onClick={[Function]}
      >
        Submit Data
      </NavLink>
    </Nav>
  </NavbarCollapse>
</Navbar>
`;

exports[`Header renders the nav bar with a brand, toggle, and collapse 1`] = `
<Navbar
  bg="light"
  collapseOnSelect={false}
  expand="lg"
  fixed="top"
  variant="light"
>
  <NavbarBrand
    onClick={[Function]}
  >
    <img
      alt="logo"
      src={null}
      style={
        Object {
          "height": "20px",
          "marginTop": "-8px",
        }
      }
    />
  </NavbarBrand>
  <NavbarToggle
    aria-controls="responsive-navbar-nav"
    label="Toggle navigation"
  />
  <NavbarCollapse
    id="responsive-navbar-nav"
  >
    <Nav
      className="mr-auto"
      fill={false}
      justify={false}
    >
      <NavLink
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "SafeAnchor",
            "render": [Function],
          }
        }
        className="active"
        disabled={false}
        id="home-tab"
        onClick={[Function]}
      >
        Home
      </NavLink>
      <NavLink
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "SafeAnchor",
            "render": [Function],
          }
        }
        className=""
        disabled={false}
        id="volunteer-tab"
        onClick={[Function]}
      >
        Volunteer
      </NavLink>
      <NavLink
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "SafeAnchor",
            "render": [Function],
          }
        }
        className=""
        disabled={false}
        id="aid-tab"
        onClick={[Function]}
      >
        Request Aid
      </NavLink>
      <NavLink
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "SafeAnchor",
            "render": [Function],
          }
        }
        className=""
        disabled={false}
        id="submit-tab"
        onClick={[Function]}
      >
        Submit Data
      </NavLink>
    </Nav>
  </NavbarCollapse>
</Navbar>
`;
